climakitae/explore/amy.py

```diff
@@ -203,6 +296,574 @@ def _return_diurnal(y: xr.DataArray) -> xr.DataArray:
+            return sim_str.split("_")[0] if "_" in sim_str else sim_str
+
+    # Process all data using quantile computation across years
+    print(
+        f"      ⚙️  Computing quantiles for {len(warming_levels)} warming level(s) and {n_simulations} simulation(s)"
+    )
+
+    # Initialize storage for profiles
+    profile_data = {}
+
+    # Progress tracking
+    total_combinations = len(warming_levels) * n_simulations
+    with tqdm(
+        total=total_combinations,
+        desc="      Computing profiles",
+        unit="combo",
+        leave=False,
+    ) as pbar:
+
+        for wl_idx, wl in enumerate(warming_levels):
+            for sim_idx, sim in enumerate(simulations):
+                # Get simulation label
+                sim_label = _get_simulation_label(sim, sim_idx)
+
+                # Select data for this warming level and simulation combination
+                if has_simulation:
+                    subset_data = data.isel(warming_level=wl_idx, simulation=sim_idx)
+                else:
+                    subset_data = data.isel(warming_level=wl_idx)
+
+                # Group by hour_of_year and find the actual data value closest to the quantile
+                # This gives us the actual data point closest to the q-th quantile for each of the 8760 hours
+                # Load data to avoid dask chunking issues with quantile
+                if hasattr(subset_data.data, "chunks"):
+                    # If it's a dask array, load it into memory
+                    subset_data = subset_data.compute()
+
+                def _closest_to_quantile(dat: xr.DataArray) -> xr.DataArray:
+                    """Find the actual data value closest to the specified quantile."""
+                    # Stack all dimensions except time_delta into a single dimension
+                    stacked = dat.stack(all_dims=list(dat.dims))
+                    # Compute the target quantile value
+                    target_quantile = stacked.quantile(q, dim="all_dims")
+                    # Find the index of the value closest to the quantile
+                    closest_idx = abs(stacked - target_quantile).argmin(dim="all_dims")
+                    # Return the actual data value at that index
+                    return xr.DataArray(stacked.isel(all_dims=closest_idx).values)
+
+                profile_1d = subset_data.groupby("hour_of_year").map(
+                    _closest_to_quantile
+                )
+
+                # Reshape to (days_in_year, 24) for the final DataFrame
+                profile_reshaped = profile_1d.values.reshape(
+                    days_in_year, hours_per_day
+                )
+
+                # Store the profile
+                key = (f"WL_{wl}", sim_label)
+                profile_data[key] = profile_reshaped
+
+                pbar.update(1)
+
+    # Create the multi-index DataFrame structure
+    hours = np.arange(1, 25, 1)  # Hours 1-24
+
+    if len(warming_levels) == 1 and n_simulations == 1:
```

Comment:
Also could be nice to split this function from here on, since before this is compute the profile, and after this is formatting the computed profiles